presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduClientSession.java

```diff
@@ -543,7 +543,11 @@ private KuduPredicate createComparisonPredicate(ColumnSchema columnSchema, KuduP
         io.prestosql.spi.type.Type type = TypeHelper.fromKuduColumn(columnSchema);
         Object javaValue = TypeHelper.getJavaValue(type, value);
         if (javaValue instanceof Long) {
-            return KuduPredicate.newComparisonPredicate(columnSchema, op, (Long) javaValue);
+            if (type.getClass().getSuperclass().equals(DecimalType.class)) {
```

Comment:
Can we use `instanceof` here ? 
```suggestion
            if (type instanceof DecimalType) {
```

Reply:
You are right! I will do that!